src/lib/core/utils/data-element-constraints.ts

```diff
@@ -215,38 +221,55 @@ export type VariablesByInputName = Partial<Record<string, VariableDescriptor>>;
  *
- * If the user has already chosen a string-type xAxisVariable, the
- * constraints.filter() below will allow constraints[0] to pass but
- * will exclude constraints[1] because the already chosen string
- * x-variable is not a date. It won't even check the yAxisVariable of
- * constraint[1] because of the all-or-nothing nature of constraints.
+ * If the user has already chosen a string-type xAxisVariable with 7 values, the
+ * constraints.filter() below will allow only constraints[0] to pass for the yAxisVariable. 
+ * The xAxisVariable is already chosen and so restricts the yAxisVariable to constraints that satisfy
+ * the xAxisVariable choice. It won't even check the yAxisVariable of
+ * constraint[1] because of the all-or-nothing nature of constraints. However, we should still be 
+ * able to switch the xAxisVariable to a date or var with < 5 values, so for the xAxisVariable we 
+ * will pass constraints[0:2]. 
+ * 
+ * Additionally, we do not want to merge constraints because, for example, that would
+ * squeeze the xAxisVariables, with no variables chosen, to only allow vars that are strings or dates *and* 
+ * have fewer than five values.
  *
- * The constraints passed by the filter are merged into one.
- *
- * If no variable has been selected by the user, then the final merged constraint would be
- * { xAxisVariable: { allowedTypes: ['string','date'] }, yAxisVariable: { allowedTypes: ['number', 'string'] } }
+ * More exampels:
+ * If no variable has been selected by the user, then all constraints will be used for
+ * both the x and y variables.
  *
+ * If the yAxisVariable has been set to a string, the xAxisVariable will be restricted to 
+ * constraints[1:2], while the yAxisVariable will use all three constraints, since it is
+ * the only variable chosen
+ * 
+ * If the xAxisVariable is a date with 4 unique values, the xAxisVariable will use all three constraint patterns
+ * while the yAxisVariable will only use constraints[1:2].
+ * 
  */
-export function flattenConstraints(
+
+export function filterConstraints(
   variables: VariablesByInputName,
   entities: StudyEntity[],
-  constraints: DataElementConstraintRecord[]
-): DataElementConstraintRecord {
+  constraints: DataElementConstraintRecord[],
+  selectedVarReference: string // variable reference for which to determine constraints. Ex. xAxisVariable.
+): DataElementConstraintRecord[] {
   // Find all compatible constraints
   const compatibleConstraints = constraints.filter((constraintRecord) =>
     Object.entries(constraintRecord).every(([variableName, constraint]) => {
       const value = variables[variableName];
       // If a value (variable) has not been user-selected for this constraint, then it is considered to be "in-play"
       if (value == null) return true;
+      // Ignore constraints that are on this selectedVarReference
+      if (selectedVarReference === variableName) return true;
```

Comment:
I wasn't able to talk to @asizemore about whether this logic fixed a particular bug. Since it seems to work, we can leave it and readdress it in the future if it causes issues.